src/wp-includes/plugin.php

```diff
@@ -39,6 +39,38 @@
 if ( ! isset( $wp_current_filter ) )
 	$wp_current_filter = array();
 
+/**
+ * Check that the callback parameters match the number passed to add_action() or add_filter().
+ *
+ * @since 1.3.0
+ *
+ * @param $function_name   add_action() or add_filter(), as returned by __FUNCTION__.
+ * @param $function_to_add The callback to check.
+ * @param $accepted_args   The number of arguments claimed.
+ */
+function _check_callback_params($function_name, $function_to_add, $accepted_args)
+{
+    if ( defined( 'CP_STRICT' ) && CP_STRICT ) {
+        if ( is_string( $function_to_add ) && function_exists( $function_to_add ) ) {
+            $rf = new ReflectionFunction( $function_to_add );
+            $params = $rf->getParameters();
+            $numParams = count( $params );
+            if ( $numParams == $accepted_args ) {
+                // we're all good
+            } elseif ( $params < $accepted_args ) {
+                _doing_it_wrong( $function_name, 'The callback takes fewer parameters than the filter.', '1.3.0' );
```

Comment:
Here could we generate a `WP_Error` message and pass that through to `_doing_it_wrong()`?

Reply:
I don't think `_doing_it_wrong` is the right thing at all (this is all very early in bringing up CP so the html output makes a mess) - `WP_Error` definitely is, but I think this should exit. Maybe just `wp_die(new WP_Error(...));`?

Reply:
Yup I was totally thinking the same - some exit here would be better and passing an error message to it. I did consider suggesting we do a `Throw new` with a custom exception but figured that might be overkill since we are intending to prevent a fatal being thrown.